### ISSUE TITLE
tap2rpm can't handle .tac files outside the current directory

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#4088]
|<img alt="TimAllen's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| TimAllen reported|
|-|-|
|Trac ID|trac#4088|
|Type|enhancement|
|Created|2009-11-12 06:12:25Z|
|Branch|https://github.com/twisted/twisted/tree/tap2rm-path-4088|

If I have a .tac file in the current directory, I can package it easily enough:
```
tap2rpm -t foo.tac
```
But if the .tac file is in a different directory, things don't work:
```
# Doesn't work
tap2rpm -t resources/foo.tac
```
The .spec file that tap2rpm produces is hard-coding the filename in an awkward way, causing rpm to choke:
```
[... other output ...]
+ cp resources/foo.tac /var/tmp/twisted-foo-1.0-root/etc/twisted-taps/
cp: cannot stat `resources/foo.tac': No such file or directory
error: Bad exit status from /var/tmp/rpm-tmp.42424 (%install)
```

Attachments:

* [tap2rpm-path-handling-4088.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/185/185ab9ebc070b0b242472a9489383866a9255d5f/55c52843f1547a14b1d8f0c461b84329f95d1239.patch) (916 bytes) - added by Screwtape on 2010-03-01 10:49:38Z - Test that tap2rpm handles .tap files outside the current directory.

<details><summary>Searchable metadata</summary>

```
trac-id__4088 4088
type__enhancement enhancement
reporter__TimAllen TimAllen
priority__normal normal
milestone__ 
branch__branches_tap2rm_path_4088 branches/tap2rm-path-4088
branch_author__therve therve
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1258006345000000 1258006345000000
changetime__1267462160000000 1267462160000000
version__None None
owner__ 
cc__therve
```
</details>

**Comment by twisted-trac** [comment]
|<img alt="TimAllen's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_1"></a>|TimAllen commented|
|-|-|

This ticket split off from [#3292](https://github.com/twisted/twisted/issues/3292).

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_2"></a>|@therve set owner to TimAllen|
|-|-|

It might be fixed in r28572. Could you write a test case for it, Tim? Thanks!

**Comment by twisted-trac** [comment]
|<img alt="Screwtape's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_3"></a>|Screwtape removed owner|
|-|-|

The attached patch against today's SVN trunk passes on my home machine (Ubuntu 9.10, Python 2.6, RPM 4.7.0) although because r28572 is already merged I can't be exactly sure that it would have failed with the previous version of `tap2rpm`.

tap2rpm takes the tapfile from the command-line, and packages it in a source tarball for RPM to read. I believe the previous version of tap2rpm was using putting the tapfile into the tarball at some path A, but putting the original filesystem path B into the RPM .spec file. Hence, RPM couldn't find the tapfile unless B happened to exactly match A. If I hack the new tap2rpm to make the same parameter/basename confusion:
```
diff --git a/twisted/scripts/tap2rpm.py b/twisted/scripts/tap2rpm.py
index 4b2ba8e..652c8ad 100755
--- a/twisted/scripts/tap2rpm.py
+++ b/twisted/scripts/tap2rpm.py
@@ -263,7 +263,7 @@ def setupBuildFiles(buildDir, config):
             'rpm_file': config['rpmfile'],
             'version': config['set-version'],
             'tarfile_basename': tarballName,
-            'tap_file': tapFileBase,
+            'tap_file': config['tapfile'],
             'date': time.strftime('%a %b %d %Y', time.localtime(time.time())),
             'maintainer': config['maintainer'],
             'long_description': config['long_description'],
```
...the attached test fails in exactly the way described in comment 0, so I believe this is a correct test.

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_4"></a>|@therve commented|
|-|-|

(In [28591]) Branching to 'tap2rm-path-4088'

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_5"></a>|@therve commented|
|-|-|

(In [28592]) Apply patch with test

Refs [#4088](https://github.com/twisted/twisted/issues/4088)

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)|@therve set owner to @therve|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_7"></a>|@therve set status to `closed`|
|-|-|

(In [28594]) Merge tap2rm-path-4088

Author: TimAllen
Reviewer: therve
Fixes [#4088](https://github.com/twisted/twisted/issues/4088)

Add a test for verifying that tap2rpm handles correctly tac files outside of
the current directory.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
